processes/consumer/flush.go

```diff
@@ -85,35 +89,46 @@ func Flush(ctx context.Context, inMemDB *models.DatabaseData, dest destination.B
 				"reason":   args.Reason,
 			}
 
-			var err error
-			action := "merge"
-			// Merge or Append depending on the mode.
-			if _tableData.Mode() == config.History {
-				err = dest.Append(_tableData.TableData, false)
-				action = "append"
-			} else {
-				err = dest.Merge(_tableData.TableData)
-			}
+			err = retry.WithRetries(retryCfg, func(_ int, _ error) error {
```

Comment:
If you capture `attempts` and store it in a variable you could add it to `tags` so we can track attempts in Datadog.

Reply:
I thought about that too, but we can get the attempts into Datadog through the logging statement above. 

This would then not incur additional unique metric count. We also don't need granularity of the attempts at the Kafka topic config level as it's going to be used on aggregate to inform us on the retry configs.